store/src/java/com/zimbra/cs/event/SolrEventStore.java

```diff
@@ -174,39 +180,49 @@ private String getAggregationBucketForContactFrequencyGraphTimeRange(ContactAnal
-        LocalDateTime now = LocalDateTime.now();
-        LocalDateTime firstDayOfMonth = now.with(TemporalAdjusters.firstDayOfMonth()).truncatedTo(ChronoUnit.DAYS);
-        return firstDayOfMonth;
+    private ZonedDateTime getStartDateForCurrentMonth(ZoneId userZoneId) {
+        ZonedDateTime userTZNow = ZonedDateTime.now(userZoneId);
+        ZonedDateTime userTZFirstDayOfMonth = userTZNow.with(TemporalAdjusters.firstDayOfMonth()).truncatedTo(ChronoUnit.DAYS);
+        return userTZFirstDayOfMonth;
     }
 
-    private LocalDateTime getStartDateForLastSixMonths() throws ServiceException {
+    private ZonedDateTime getStartDateForLastSixMonths(ZoneId userZoneId) throws ServiceException {
         //attr id="261" name="zimbraPrefCalendarFirstDayOfWeek". sunday = 0...saturday = 6
         int firstDayOfWeek = Provisioning.getInstance().getAccountById(accountId).getPrefCalendarFirstDayOfWeek();
         //As per "zimbraPrefCalendarFirstDayOfWeek" sunday = 0...saturday = 6. But WeekFields takes days as sunday = 1....saturday = 7
         int firstDayOfWeekAdjusted = firstDayOfWeek + 1;
-        LocalDateTime firstDayOfCurrentWeekAsConfigured = LocalDateTime.now().with(WeekFields.of(Locale.US).dayOfWeek(), firstDayOfWeekAdjusted).truncatedTo(ChronoUnit.DAYS);
-        LocalDateTime firstDayOfWeek6MonthsBack = firstDayOfCurrentWeekAsConfigured.minusMonths(6).with(WeekFields.of(Locale.US).dayOfWeek(), firstDayOfWeekAdjusted);
-        return firstDayOfWeek6MonthsBack;
+        ZonedDateTime userTZFirstDayOfCurrentWeekAsConfigured = ZonedDateTime.now(userZoneId).with(WeekFields.of(Locale.US).dayOfWeek(), firstDayOfWeekAdjusted).truncatedTo(ChronoUnit.DAYS);
+        ZonedDateTime userTZFirstDayOfWeek6MonthsBack = userTZFirstDayOfCurrentWeekAsConfigured.minusMonths(6).with(WeekFields.of(Locale.US).dayOfWeek(), firstDayOfWeekAdjusted);
+        return userTZFirstDayOfWeek6MonthsBack;
+    }
+
+    private ZonedDateTime getStartDateForCurrentYear(ZoneId userZoneId) {
+        ZonedDateTime userTZNow = ZonedDateTime.now(userZoneId);
+        ZonedDateTime userTZFirstDayOfCurrentYear = userTZNow.with(TemporalAdjusters.firstDayOfYear()).truncatedTo(ChronoUnit.DAYS);
+        return userTZFirstDayOfCurrentYear;
     }
 
-    private LocalDateTime getStartDateForCurrentYear() {
-        LocalDateTime now = LocalDateTime.now();
-        LocalDateTime firstDayOfCurrentYear = now.with(TemporalAdjusters.firstDayOfYear()).truncatedTo(ChronoUnit.DAYS);
-        return firstDayOfCurrentYear;
+    private String getSolrTimeZone(Integer offsetInMinutes) {
+        String sign = offsetInMinutes >= 0 ? "+" : "-";
+        int hours = Math.abs(offsetInMinutes) / 60;
+        int minutes = Math.abs(offsetInMinutes) % 60;
+        String timeFormat = "%02d";
+        StringBuilder solrTimeZone = new StringBuilder();
+        solrTimeZone.append("GMT").append(sign).append(String.format(timeFormat, hours)).append(String.format(timeFormat, minutes));
```

Comment:
This string has no conditionals, so you could build this string with String.format() without mixing in StringBuilder calls: `String.format("GMT%s%02d%02d", sign, hours, minutes)`